k9mail/src/main/java/com/fsck/k9/ui/crypto/MessageCryptoHelper.java

```diff
@@ -79,7 +83,9 @@
     private CryptoPart currentCryptoPart;
     private Intent currentCryptoResult;
     private Intent userInteractionResultIntent;
+    private boolean firstPassStarted;
```

Comment:
Why not use a state enum instead of three booleans? Also, naming the states instead of using numbers can give an idea what actions are performed without having to read all of the code.